### ISSUE TITLE
MD034 Bare URL used should be ignored in Code segments

### ISSUE THREAD

**Issue by JanoschDeurer** [markdownlint/markdownlint#191]
Markdown code like
````markdown
```bash
git clone https://github.com/markdownlint/markdownlint.git
```
````
should not raise MD034 as inside code blocks this is completely fine. I think this could affect other rules too.

**Comment by shichao-an** [comment]
Same thing if a normal link in HTML tags, e.g `<u></u>`.

```
<u>Currency in the United States is issued by the [U.S. Treasury](https://en.wikipedia.org/wiki/United_States_Department_of_the_Treasury) and circulated by the Fed.</u>
```

This raises `MD034 Bare URL used`. If `<u></u>` is removed, it won't raise.

**Comment by gwern** [comment]
Yes, I ran into the same issue with `<div>` tags as well:

```
$ mdl --version
0.8.0
 $ cat foo.page
<div>
[link](http://www.example.com)
$ mdl foo.page 
foo.page:2: MD034 Bare URL used

A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

I use div tags a lot for supporting nice format features like epigraphs and abstracts, so this bug gets triggered a lot when I run `mdl` on my files.

**Comment by jaymzh** [comment]
I think these are two different cases, but both are indeed bugs. Thanks.

**Comment by jasonk** [comment]
Should also be ignored in a "reference-style link":
```
See the docs about [references-style-links][reflinks].

[reflinks]: https://www.markdownguide.org/basic-syntax/#reference-style-links
```

**Comment by clark-ja** [comment]
@jaymzh any news on this? (bare URLs in code blocks)

**Comment by jaymzh** [comment]
@theguy147 sorry, in the last 4 months, I've had major back surgery, my mother passed away and I've been trying to find long-term care for my father... I'm happy to review PRs, but I haven't had time to work on the code myself.

**Comment by claremacrae** [comment]
So very sorry to hear that…

**Comment by jaymzh** [comment]
This was fixed in 0.16.0
